### DISCUSSION TITLE
Have a problem in mapping an object

### DISCUSSION THREAD

**Discussion by erfan2220** [vercel/next.js#47599] [Help]
### Summary

HI.
This is my error: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: array. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

I know I can't map an object but I need another way.
When I want to map an object I got this error .

**Comment by icyJoseph** [comment]
Hi,

Could you share some code? pseudocode at least, so that we don't make any assumptions?

A way to map over an the key-value pairs of an object is to use `Object.entries(yourObjectGoesHere).map(callbackGoesHere)`, but I don't think that's the issue you are having, it sounds like you have:

```jsx
import "./styles.css";
import { createElement } from "react";

export default function App() {
  return (
    <div>
     {createElement([])}
    </div>
  );
}
```

**Reply by icyJoseph** [reply]
@erfan2220 did you sort out the issue? I received a notification with a link to a repository, but it gives 404.

**Reply by erfan2220** [reply]
https://github.com/erfan2220/testadditem
this is link of project
I have error and I can't resolve that

**Reply by icyJoseph** [reply]
Hi @erfan2220, the problem is that you are doing variable shadowing.

For example in:

```js
let x = 0;

function doSomething() {
   let x = 100; // shadows the let x = 0; declaration above
   console.log(x);
}

doSomething()
console.log(x); // no longer shadowed
```

The top level `x` is shadowed by the `doSomething` - `x`, so everything after `let x = 100`, takes the 100 value, but only until the doSomething function ends, then you'd go back to zero value.

In your case specifically:

```jsx
import List from "./List";
import {useState} from "react";

function App() {
   // other state setters
   let [List,setList]=useState([]) // here you are shadowing import List from "./List";
   // List is no longer a component, but an array


  // some more code goes here
  return (
    <section className="list-it">
      <form className="grocery-form" onSubmit={Handlesubmit}>
          {alert.show && <Alert />}
        <input type="text" placeholder="this is for writing list" value={Name} onChange={(e)=>{setName(e.target.value)}}/>
        <button className="form-btn" type="submit"> {Isediting?"edit":"submit"} </button>
      </form>
        {/* Here you are now trying to render <[] items={[]} /> , because List is an array*/}
        <List items={List}/>
      <button className="clear-btn">clear button</button>
    </section>
  );
}
``` 

You could fix it quickly by changing your state `List` to `list`, with lowercased `l`, like so:

```jsx
 let [list,setList]=useState([])

// later on 

        <List items={list}/>
```

I'd recommend giving things more meaningful, perhaps longer, names, and more importantly, use ESLint to forbid shadowing.  https://eslint.org/docs/latest/rules/no-shadow

**Reply by erfan2220** [reply]
Thanks for your reply.It works

**Reply by icyJoseph** [reply]
Nice!, Please, don't forget to mark as answered ✨

**Reply by erfan2220** [reply]
@icyJoseph You are great man.Thanks
